frontend/controller/actions/group.js

```diff
@@ -527,16 +506,21 @@ export default (sbp('sbp/selectors/register', {
     // We can avoid this by waiting on both contracts, especially the group
     // contract.
     await sbp('chelonia/contract/wait', [groupId, identityContractID])
-    await sbp('gi.actions/identity/joinGroup', {
-      contractID: identityContractID,
-      contractName: 'gi.contracts/identity',
-      data: {
-        groupContractID: groupId,
-        inviteSecret: secret
-      }
-    }).then(() => {
-      return sbp('gi.actions/group/switch', groupId)
-    })
+    try {
+      await sbp('chelonia/contract/retain', groupId, { ephemeral: true })
+      await sbp('gi.actions/identity/joinGroup', {
+        contractID: identityContractID,
+        contractName: 'gi.contracts/identity',
+        data: {
+          groupContractID: groupId,
+          inviteSecret: secret
+        }
+      }).then(() => {
+        return sbp('gi.actions/group/switch', groupId)
+      })
+    } finally {
+      await sbp('chelonia/contract/release', groupId, { ephemeral: true })
+    }
```

Comment:
Why are we calling retain and release on the `groupId` here?

Shouldn't we just wait for the sideEffect of `joinGroup` to get us to join the group?

Reply:
This replaces the old `JOINING_GROUP` globals. We can't (just) wait for the side-effect because we may be re-joining, and the old leave action may remove the group. This `retain`/`release` ensures that the group is kept around until we know what to do next.

Reply:
(Compare this to the process for joining a chatroom, which uses the same pattern and for the same reason)

Reply:
> What happens if this block gets executed before the old leave action is processed? Won't it have zero effect in that case?

It won't be executed in its entirety before the old leave action is processed because sending the `joinGroup` action (or any action) requires being up to date with the contract. However, if it somehow happened, it'd have 0 effect as you say, which is still correct.